public/app/features/plugins/components/AppRootPage.tsx

```diff
@@ -81,16 +82,56 @@ export function AppRootPage({ pluginId, pluginNavSection }: Props) {
     />
   );
 
+  // Because of the fallback at plugin routes, we need to check
+  // if the user has permissions to see the plugin page.
+  const userHasPermissionsToPluginPage = (path: string) => {
+    if (!plugin.meta?.includes) {
+      return true;
+    }
+    let rolesMap = new Map();
+    plugin.meta?.includes.map((include) => {
+      rolesMap.set(include.path, include.role);
+    });
+    if (contextSrv.isGrafanaAdmin) {
+      return true;
+    }
```

Comment:
I think this `isGrafanaAdmin` check could go to the top of the function (if it's an admin we don't need to perform any other logic).